### ISSUE TITLE
Severe limitation on managing NPM modules via Solution Explorer

### ISSUE THREAD

**Issue by jitterbox** [MicrosoftDocs/visualstudio-docs#3399]
NPM modules **cannot be managed in Solution Explorer** VS 2017/2019 if the **node_modules** folder is deeper than the solution root.  Unfortunately, the ASP.NET Core template for Angular projects does just this by placing it under the /ClientApp folder.  This has been raised in multiple issues but I don't believe it has been/will be fixed by any of the MS teams involved.  If it remains an issue, I believe this limitation deserves to be noted in the docs.
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 25963783-6129-cbef-174c-9dd4adde494b
* Version Independent ID: 85926796-ac02-c41b-8ffc-c8987e22f387
* Content: [Manage npm packages - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/javascript/npm-package-management?view=vs-2019#solutionExplorer)
* Content Source: [docs/javascript/npm-package-management.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/javascript/npm-package-management.md)
* Product: **visual-studio-windows**
* Technology: **vs-javascript**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

**Comment by Mikejo5000** [comment]
@jitterbox Thanks for the info. I'll plan to add a note about that.

**Comment by Mikejo5000** [comment]
@armanio123 Is the location of node_modules configurable through the properties?

**Comment by armanio123** [comment]
@Mikejo5000 I believe there are some ways of changing the node_modules path but is not Visual Studio specific configuration. Also, I understand the limitation is on the NPM node itself which requires to have a `package.json` on the root path of the project.

**Comment by Mikejo5000** [comment]
OK, thx, Armando @jitterbox I don't know if this will meet your requirements for VS integration, but I created and ran the .NET Core template as described here:
https://docs.microsoft.com/en-us/aspnet/core/client-side/spa/angular?view=aspnetcore-2.2&tabs=visual-studio
Then I closed the project and opened it as a folder, as described here:
https://docs.microsoft.com/en-us/visualstudio/javascript/develop-javascript-code-without-solutions-projects?view=vs-2019
At this point I could use instructions in the specified article to manage npm packages using VS.
I could also run the app by right-clicking on project.json, choosing Npm > npm run-script start,
then load the page as described in the console window (i.e., opening http://localhost:4200/), which worked.
I'm not sure what all the limitations might be with this approach.
